File: task3/.cpp
#include <iostream>
using namespace std;

bool isAscending(int num) {
if (num < 10) return false;
int prev = num % 10;
num /= 10;
while (num > 0) {
int curr = num % 10;
if (curr >= prev) return false;
prev = curr;
num /= 10;
}
return true;
}

int reverseNumber(int num) {
int reversed = 0;
while (num > 0) {
reversed = reversed * 10 + num % 10;
num /= 10;
}
return reversed;
}

int main() {
int n, num;
cin >> n;

for (int i = 0; i < n; i++) {
cin >> num;
if (isAscending(num)) {
cout << reverseNumber(num) << " ";
}
}

return 0;
}

File: task3/.go
package main

import (
	"fmt"       
	"strconv" 
)
func isAscending(n int) bool { 
	s := strconv.Itoa(n) // Преобразуем число в строку
	for i := 1; i < len(s); i++ { // Цикл по строке, начиная со второго символа
		if s[i] <= s[i-1] { 
			return false // Число не возрастающее
		}
	}
	return true // Число возрастающее
}
func reverseNumber(n int) int { 
	s := strconv.Itoa(n) // Преобразуем число в строку
	runes := []rune(s) // Преобразуем строку 
	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 { // Переворачиваем 
		runes[i], runes[j] = runes[j], runes[i] // Меняем местами символы
	}
	reversedS := string(runes) // Преобразуем обратно в строку
	reversedN, err := strconv.Atoi(reversedS) // Преобразуем строку обратно в число
	if err != nil { // Если произошла ошибка при преобразовании
		return 0 
	}
	return reversedN 
}
func main() { 
	numbers := []int{4, 87, 129, 33, 45} 
	fmt.Print("Result4: ") 
	for _, num := range numbers { 
		if isAscending(num) { // Если число возрастающее
			fmt.Printf("%d ", reverseNumber(num)) 
		}
	}
	fmt.Println() 
}
